pkg/workload/workload.go

```diff
@@ -236,10 +243,63 @@ func BaseSSAWorkload(w *kueue.Workload) *kueue.Workload {
 	return wlCopy
 }
 
-// AdmissionPatch creates a new object based on the input workload that
-// contains the admission. The object can be used in Server-Side-Apply.
-func AdmissionPatch(w *kueue.Workload) *kueue.Workload {
+// SetAdmission applies the provided admission to the workload.
+// The WorkloadAdmitted and WorkloadEvicted are added or updated if necessary.
+func SetAdmission(w *kueue.Workload, admission *kueue.Admission) {
```

Comment:
How about we keep the `Set` and `Unset` functions but make it so that they don't call` Patch`. Then the function that actually calls Patch with SSA can be called `Apply`.

Reply:
See if you can accommodate this in this PR. If it's a lot of changes, then we can do it in a follow up.

Reply:
done , for the admissin related changes , only `ApplyAdmissionStatus` will do k8s object changes. 

The is also `UpdateStatus` which is doing both, if needed this should be done in another PR.